src/router.rs

```diff
@@ -143,10 +143,15 @@ impl Handler for C {
         let C(f) = *self;
         match f(req) {
             Ok(resp) => Ok(resp),
-            Err(e) => match e.response() {
-                Some(response) => Ok(response),
-                None => Err(std_error(e)),
-            },
+            Err(e) => {
+                if let Some(cause) = e.cause() {
+                    req.log_metadata("cause", cause.to_string())
```

Comment:
This will not log nested causes (like error -> cause -> cause). Is this something we expect to happen?

Reply:
This actually already works!  I've added a commit to demonstrate that in the test.